optlang/gurobi_interface.py

```diff
@@ -371,22 +373,35 @@ def get_linear_coefficients(self, variables):
     def _get_expression(self):
         if self.problem is not None and self._expression_expired and len(self.problem._variables) > 0:
             grb_obj = self.problem.problem.getObjective()
-            terms = []
             variables = self.problem._variables
-            for i in range(grb_obj.size()):
-                terms.append(grb_obj.getCoeff(i) * variables[grb_obj.getVar(i).getAttr('VarName')])
-            expression = symbolics.add(terms)
-            # TODO implement quadratic objectives
-            self._expression = expression + getattr(self.problem, "_objective_offset", 0)
+            if self.problem.IsQP:
```

Comment:
I think you're missing a `problem` here? `if self.problem.problem.IsQP`?

Reply:
done